src/core/Provider.ts

```diff
@@ -30,6 +41,11 @@ export const Provider: React.FC<{
       storeRef.current as ReturnType<typeof createStore>,
       atomsRef.current
     )
+    children = createElement(
+      RegisteredAtomsContext.Provider,
+      { value: atomsRef.current },
```

Comment:
So, because this doesn't trigger useAtomsSnapshot to re-render, we can't capture atoms created on demand?